core/domain/learner_progress_services.py

```diff
@@ -616,3 +648,118 @@ def get_collection_summary_dicts(collection_summaries):
         })
 
     return summary_dicts
+
+
+def get_activity_progress(user_id):
+    """Returns the progress of the learners - the explorations and collections
+    completed by the user and those in progress.
+
+    Args:
+        user_id: str. The id of the learner.
+
+    Returns:
```

Comment:
I have combined the summaries into a single domain object. The deleted numbers and completed to incomplete collections I have left separate because they are more like peripherals and not really part of the learner progress.